### PR TITLE
(#12365): Jungle recipe

### LINKED ISSUE #12365: [request] jungle/cci.20220801
### Package Details
  * Package Name/Version: **jungle/cci.20220801**
  * Website: **https://github.com/eBay/Jungle**
  * Source code: **https://github.com/eBay/Jungle**

### Description Of The Library / Tool
Embedded key-value storage library based on LSM and [copy-on-write (append-only) B+tree ForestDB.

### PR DESCRIPTION
**Pull request by szmyd**
Specify library name and version:  **jungle/cci.20220801**

Jungle is a KV-Store based on the LSM and CoW B+Tree from ForestDB.

closes #12365

---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/jungle/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
```
Comment (uilianries) [inline recipes/jungle/all/conanfile.py:2]:
```suggestion
```
Unused

File: recipes/jungle/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+from conan.tools.files import get, patch
+from conans import CMake
+from conans.tools import check_min_cppstd
+
+required_conan_version = ">=1.33.0"
```
Comment (uilianries) [inline recipes/jungle/all/conanfile.py:7]:
```suggestion
required_conan_version = ">=1.50.0"
```
 from conans.tools import check_min_cppstd requires 1.50

File: recipes/jungle/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def build(self):
+        for patch_file in self.conan_data.get("patches", {}).get(self.version, []):
+            patch(self, **patch_file)
```
Comment (uilianries) [inline recipes/jungle/all/conanfile.py:55]:
```suggestion
        apply_conandata_patches(self)
```
https://docs.conan.io/en/latest/reference/conanfile/tools/files/patches.html#conan-tools-files-apply-conandata-patches

File: recipes/jungle/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import get, patch
```
Comment (uilianries) [inline recipes/jungle/all/conanfile.py:3]:
```suggestion
from conan.tools.files import get, patch, apply_conandata_patches
```

File: recipes/jungle/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="LICENSE*", src=self._source_subfolder, dst="licenses")
+        self.copy("*.a*", dst="lib", src="lib")
+        self.copy("*.lib", dst="lib", src="lib")
+        self.copy("*.so*", dst="lib", src="lib", symlinks=True)
+        self.copy("*.dylib*", dst="lib", src="lib", symlinks=True)
+        self.copy("*.dll*", dst="lib", src="lib")
```
Comment (uilianries) [inline recipes/jungle/all/conanfile.py:66]:
```suggestion
        self.copy("*.dll*", dst="bin", src="lib")
```
DLL files should be copied to bin folder.

File: recipes/jungle/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        self.copy("*.h", dst="include/", src="%s/include" % (self._source_subfolder), keep_path=True)
```
Comment (uilianries) [inline recipes/jungle/all/conanfile.py:67]:
```suggestion
        self.copy("*.h", dst="include", src=os.path.join(self._source_subfolder, "include"), keep_path=True)
```
Prefer os.path.join always due os.sep

File: recipes/jungle/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+    def package_info(self):
+        self.cpp_info.libs = ["jungle"]
+        self.cpp_info.system_libs.extend(["pthread", "dl"])
```
Comment (uilianries) [inline recipes/jungle/all/conanfile.py:71]:
```suggestion
        if self.settings.os in ["Linux", "FreeBSD"]:
            self.cpp_info.system_libs.extend(["pthread", "dl"])
```
Only some OS need it.

File: recipes/jungle/all/test_package/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.11)
+set(CMAKE_CXX_STANDARD 11)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+find_package(jungle REQUIRED)
```
Comment (uilianries) [inline recipes/jungle/all/test_package/CMakeLists.txt:8]:
```suggestion
find_package(jungle CONFIG REQUIRED)
```

File: recipes/jungle/all/test_package/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from conans import ConanFile, CMake, tools, RunEnvironment
+from conan.tools.build import cross_building
+import os
+import subprocess
+import re
```
Comment (uilianries) [inline recipes/jungle/all/test_package/conanfile.py:8]:
```suggestion
from conan import ConanFile
from conan.tools.build import cross_building
from conans import CMake
import os
```
No she-bang. Avoid unused imports

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution.

File: recipes/jungle/all/test_package/CMakeLists.txt (hunk 8 above)
Comment (szmyd) [reply recipes/jungle/all/test_package/CMakeLists.txt:8]: This is the only review change I'm not quite sure I grok. Isn't this only needed when `cmake_find_package_multi` is used?

**Review by uilianries — DISMISSED** [review]

File: recipes/jungle/all/conandata.yml [hunk 10]
```diff
@@ -0,0 +1,12 @@
+sources:
+  "cci.20220801":
+    url: "https://github.com/eBay/Jungle/archive/f41b7123489f1bc942a6b76dc54485391485cd27.tar.gz"
+    sha256: 8667a114bcef661b2a93e627a68b0584931f182dc8b96693ce6901d903584ab8
+patches:
+  "cci.20220801":
```
Comment (uilianries) [inline recipes/jungle/all/conandata.yml:6]: Please, update patches section, adding a short description and its type, like described:  https://github.com/conan-io/conan-center-index/blob/master/docs/conandata_yml_format.md#patches

Comment (szmyd) [reply recipes/jungle/all/conandata.yml:6]: :+1:

**Review by uilianries — APPROVED** [review]

File: recipes/jungle/all/patches/0001-cmake-alterations.patch [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
++set(CMAKE_DIR ${TOP_SOURCE_DIR}/cmake)
++set(FDB_LIB_DIR ${TOP_SOURCE_DIR}/third_party/forestdb/build)
+ 
+ if (CODE_COVERAGE GREATER 0)
+     set(CMAKE_BUILD_TYPE "Debug")
+@@ -45,10 +49,10 @@ set(LIBDL dl)
+ 
+ # Includes
+ include_directories(BEFORE ./)
+-include_directories(BEFORE ${PROJECT_SOURCE_DIR}/src)
```
Comment (prince-chrismc) [inline recipes/jungle/all/patches/0001-cmake-alterations.patch:21]: Changes here seem a little much, https://cmake.org/cmake/help/latest/variable/PROJECT_SOURCE_DIR.html

> source directory of the last call to the [project()](https://cmake.org/cmake/help/latest/command/project.html#command:project) command made in

I am surprised this is required 🤔 

***

Can you confirm this was needed 🙏

**Review by prince-chrismc — COMMENTED** [review]
Just one question since I am not familiar with this project.
Otherwise its a ✅

File: recipes/jungle/all/patches/0001-cmake-alterations.patch (hunk 11 above)
Comment (szmyd) [reply recipes/jungle/all/patches/0001-cmake-alterations.patch:21]: You are correct, vestigial patch it seems.

**Review by uilianries — APPROVED** [review]
Nice piece of work! Please, consider migrating your recipe for future PRs https://docs.conan.io/en/latest/conan_v2.html

**Pull request merged**